addons/web/static/src/js/widgets/image_zoom.js

```diff
@@ -0,0 +1,109 @@
+odoo.define('web.ImageZoom', function (require) {
+    "use strict";
+
+    var Widget = require('web.Widget');
+    var basic_fields = require('web.basic_fields');
+    var core = require('web.core');
+    var registry = require('web.field_registry');
+    var session = require('web.session');
+    var field_utils = require('web.field_utils');
+
+    require("web.zoomodoo");
+
+    var qweb = core.qweb;
+    var _lt = core._lt;
+
+    function getImageUrl(model, res_id, field, unique) {
+        return session.url('/web/image', {
+            model: model,
+            id: JSON.stringify(res_id),
+            field: field,
+            unique: field_utils.format.datetime(unique).replace(/[^0-9]/g, ''),
+        });
+    }
+
+    function checkImageSize() {
+        if(this.$zoom.height() < 256 && this.$zoom.width() < 256) {
+            this.hide();
+        }
+    }
+
+    function changeFlyoutSize() {
+        this.$flyout.css({'width': '512px', 'height': '512px'});
+    }
+
+    var FieldImageWithZoom = basic_fields.FieldBinaryImage.extend({
+        description: _lt("Image with zoom"),
+
+        _render: function () {
+            this._super.apply(this, arguments);
+
+            if (this.mode === 'readonly') {
+                var url = getImageUrl(this.model, this.res_id, 'image', this.recordData.__last_update);
+
+                var $img = this.$el.find('img');
+                $img.attr('data-zoom', 1);
+                $img.attr('data-zoom-image', url);
+
+                $img.zoomOdoo({
+                    event: 'mouseenter',
+                    attach: '.o_content',
+                    attachToTarget: true,
+                    onShow: checkImageSize,
+                    beforeAttach: changeFlyoutSize,
+                });
+            }
+        }
+    });
+
+    var BackgroundImage = Widget.extend({
```

Comment:
Why is it derived from `Widget` instead of `AbstractField`and why is it in a different class? 

Couldn't it be merged with `FieldImageWithZoom` and just add an option to the field in order to trigger the behaviour of `BackgroundImage`?

Moreover merging this with `FieldImageWithZoom` will avoid rewriting code that is already written in `AbstractField` or `FieldBinaryImage`